From 4d0aee983a27bffceb71bc19ffc6dc3ba9672560 Mon Sep 17 00:00:00 2001
From: David Cohen <david.a.cohen@linux.intel.com>
Date: Wed, 5 Mar 2014 14:29:47 -0800
Subject: [PATCH 2/2] HACK: sdhci-acpi: add SDHCI_QUIRK2_TUNING_POLL to Intel
 Baytrail-T platform

Add SDHCI_QUIRK2_TUNING_POLL quirk to BYT on sdhci-acpi driver.

This is a hack. In order to get a cleaner solution, should we set the
quirk via acpi table??

Change-Id: I0b759a5c2c65a98eea629cf3b645d44df5f01310
Signed-off-by: David Cohen <david.a.cohen@linux.intel.com>
---
 drivers/mmc/host/sdhci-acpi.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

Index: linux/drivers/mmc/host/sdhci-acpi.c
===================================================================
--- linux.orig/drivers/mmc/host/sdhci-acpi.c	2014-05-20 09:38:58.019450890 -0700
+++ linux/drivers/mmc/host/sdhci-acpi.c	2014-05-20 09:39:35.973450210 -0700
@@ -119,10 +119,12 @@
 	.caps    = MMC_CAP_8_BIT_DATA | MMC_CAP_NONREMOVABLE | MMC_CAP_HW_RESET,
 	.caps2   = MMC_CAP2_HC_ERASE_SZ | MMC_CAP2_POLL_R1B_BUSY,
 	.flags   = SDHCI_ACPI_RUNTIME_PM,
+	.quirks2 = SDHCI_QUIRK2_TUNING_POLL,
 };
 
 static const struct sdhci_acpi_slot sdhci_acpi_slot_int_sdio = {
-	.quirks2 = SDHCI_QUIRK2_HOST_OFF_CARD_ON,
+	.quirks2 = SDHCI_QUIRK2_HOST_OFF_CARD_ON |
+		   SDHCI_QUIRK2_TUNING_POLL,
 	.caps    = MMC_CAP_NONREMOVABLE | MMC_CAP_POWER_OFF_CARD,
 	.flags   = SDHCI_ACPI_RUNTIME_PM,
 	.pm_caps = MMC_PM_KEEP_POWER,
@@ -130,7 +132,8 @@
 
 static const struct sdhci_acpi_slot sdhci_acpi_slot_int_sd = {
 	.flags   = SDHCI_ACPI_SD_CD | SDHCI_ACPI_RUNTIME_PM,
-	.quirks2 = SDHCI_QUIRK2_CARD_ON_NEEDS_BUS_ON,
+	.quirks2 = SDHCI_QUIRK2_CARD_ON_NEEDS_BUS_ON |
+		   SDHCI_QUIRK2_TUNING_POLL,
 };
 
 struct sdhci_acpi_uid_slot {
